AdvLoggerPkg/Library/BasePanicLibAdvancedLogger/BasePanicLibAdvancedLogger.c

```diff
@@ -0,0 +1,61 @@
+/** @file
+  An instance of the Panic Library that outputs to advanced logger.
+
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Base.h>
+#include <Library/BaseLib.h>
+#include <Library/PrintLib.h>
+#include <Library/AdvancedLoggerLib.h>
+#include <Library/PanicLib.h>
+
+//
+// Define the maximum panic message length that this library supports
+//
+#define MAX_PANIC_MESSAGE_LENGTH  0x100
+
+/**
+  Prints a panic message containing a filename, line number, and description.
+  This is always followed by a dead loop.
+
+  Print a message of the form "PANIC <FileName>(<LineNumber>): <Description>\n"
+  to the debug output device.  Immediately after that CpuDeadLoop() is called.
+
+  If FileName is NULL, then a <FileName> string of "(NULL) Filename" is printed.
+  If Description is NULL, then a <Description> string of "(NULL) Description" is printed.
+
+  @param  FileName     The pointer to the name of the source file that generated the panic condition.
+  @param  LineNumber   The line number in the source file that generated the panic condition
+  @param  Description  The pointer to the description of the panic condition.
+
+**/
+VOID
+EFIAPI
+PanicReport (
+  IN CONST CHAR8  *FileName,
+  IN UINTN        LineNumber,
+  IN CONST CHAR8  *Description
+  )
+{
+  CHAR8  Buffer[MAX_PANIC_MESSAGE_LENGTH];
+
+  if (FileName == NULL) {
+    FileName = "(NULL) Filename";
+  }
+
+  if (Description == NULL) {
+    Description = "(NULL) Description";
+  }
+
+  AsciiSPrint (Buffer, sizeof (Buffer), "PANIC [%a] %a(%d): %a\n", gEfiCallerBaseName, FileName, LineNumber, Description);
+
+  AdvancedLoggerWrite (0x80000000, Buffer, AsciiStrnLenS (Buffer, sizeof (Buffer)));

```

Comment:
```suggestion
  AdvancedLoggerWrite (DEBUG_ERROR, Buffer, AsciiStrnLenS (Buffer, sizeof (Buffer)));

```

Reply:
I used the integer value because the error bit is spec defined and I didn't want to bring in DebugLib just for that. I can leave a comment explaining that in the code if preferred.

Reply:
That would be fine.

Reply:
Added